File: algorithms/1_easy/F/00724_find_pivot_index/prefix_sum.c
int pivotIndex(const int *restrict nums, int len) {
    int suff = 0;
    for (int i = 0; i < len; i += 1) { suff += nums[i]; }

    int pref = 0;
    int cnt = 0;

     for (int i = 0; i < len; i+= 1) {
        suff -= nums[i];

        if (pref == suff) {
            return i;
        }

        pref += nums[i];
    }

    return -1;
}

File: algorithms/1_easy/F/00724_find_pivot_index/prefix_sum.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let mut suff: i32 = nums.iter().sum();
        let mut pref: i32 = 0;

        for (i, &num) in nums.iter().enumerate() {
            suff -= num;

            if pref == suff {
                return i.try_into().unwrap();
            }

            pref += num;
        }

        -1
    }
}
